src/ClpIrV1Decoder.cpp

```diff
@@ -0,0 +1,224 @@
+    );
+    for (auto const& log_event : log_events_span) {
+        message.clear();
+
+        // TODO: Replace below handlers code by an OSS decoding method once it's added in the future
+        auto const constant_handler = [&](std::string const& value, size_t begin_pos, size_t length
+                                      ) { message.append(value, begin_pos, length); };
+
+        auto const encoded_int_handler = [&](clp::ir::four_byte_encoded_variable_t value) {
+            message.append(clp::ffi::decode_integer_var(value));
+        };
+
+        auto const encoded_float_handler
+                = [&](clp::ir::four_byte_encoded_variable_t encoded_float) {
+                      message.append(clp::ffi::decode_float_var(encoded_float));
+                  };
+
+        auto const dict_var_handler
+                = [&](std::string const& dict_var) { message.append(dict_var); };
+
+        try {
+            clp::ffi::ir_stream::generic_decode_message<true>(
+                    log_event.get_logtype(),
+                    log_event.get_encoded_vars(),
+                    log_event.get_dict_vars(),
+                    constant_handler,
+                    encoded_int_handler,
+                    encoded_float_handler,
+                    dict_var_handler
+            );
+        } catch (clp::ffi::ir_stream::DecodingException const& e) {
+            SPDLOG_ERROR("Failed to decode. Error Code: {}", e.get_error_code());
+            break;
+        }
+
+        constexpr size_t cLogLevelNone{0};
+        constexpr size_t cLogLevelBeginIdx{cLogLevelNone + 1};
+        constexpr size_t cLogLevelPositionInMessages{1};
+        size_t log_level{cLogLevelNone};
+        for (size_t i = cLogLevelBeginIdx; i < cLogLevelNames.size(); ++i) {
+            if (message.substr(cLogLevelPositionInMessages).starts_with(cLogLevelNames[i])) {
+                log_level = i;
+                break;
+            }
+        }
+
+        m_ts_pattern.insert_formatted_timestamp(log_event.get_timestamp(), message);
+
+        EM_ASM(
+                { Emval.toValue($0).push([UTF8ToString($1), $2, $3, $4]); },
+                results.as_handle(),
+                message.c_str(),
+                log_event.get_timestamp(),
+                log_level,
+                begin_idx + (&log_event - log_events_span.data()) + 1
+        );
+    }
+
+    return results;
+}
+
+ClpIrV1Decoder::ClpIrV1Decoder(
+        std::unique_ptr<char const[]>&& data_buffer,
+        std::shared_ptr<clp::streaming_compression::zstd::Decompressor> zstd_decompressor,
+        clp::ir::LogEventDeserializer<clp::ir::four_byte_encoded_variable_t> deserializer
+)
+        : m_full_range_built{false},
+          m_data_buffer{std::move(data_buffer)},
+          m_zstd_decompressor{std::move(zstd_decompressor)},
+          m_deserializer{std::move(deserializer)},
+          m_ts_pattern{m_deserializer.get_timestamp_pattern()} {}
```

Comment:
1. If we default initilize `m_full_range_built` we don't have to initialize it here
2. We should initialize members in the same order they've been declared

Reply:
1. Done. 
2. Reordered the member variables declaration to match the constructor initializer list order.